myproject/blog/views.py

```diff
@@ -28,7 +31,7 @@ class AuthorProfile(generic.DetailView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         author = self.get_object()
-        context['author_posts'] = author.blogs.all().order_by("-created_at")
+        context['author_posts'] = BlogPost.objects.select_related().filter(author = author)
```

Comment:
You didn't passed anything in the select related, how this will work, explain me here with the proper example.